Userland/Libraries/LibWeb/Page/EventHandler.cpp

```diff
@@ -542,7 +542,8 @@ bool EventHandler::handle_mousemove(CSSPixelPoint position, CSSPixelPoint screen
     if (hovered_node_changed) {
         JS::GCPtr<HTML::HTMLElement const> hovered_html_element = document.hovered_node() ? document.hovered_node()->enclosing_html_element_with_attribute(HTML::AttributeNames::title) : nullptr;
         if (hovered_html_element && hovered_html_element->title().has_value()) {
-            page.client().page_did_enter_tooltip_area(m_browsing_context->active_document()->navigable()->to_top_level_position(position), hovered_html_element->title()->to_byte_string());
+            auto title = hovered_html_element->title()->bytes_as_string_view().replace("\r\n"sv, "\n"sv, ReplaceMode::All).replace("\r"sv, "\n"sv, ReplaceMode::All);
+            page.client().page_did_enter_tooltip_area(m_browsing_context->active_document()->navigable()->to_top_level_position(position), title);
```

Comment:
This feels like an arbitrary thing to do CR/LF conversion on. If we do it on link titles, what else should we be doing it on?
Would it make sense to handle this in the GUI library instead?

Reply:
My argument was that we want to normalize the line endings in the WebContent ipc to LF. All current GUI Frameworks: AppKit, Qt & LibGUI support this. When a GUI framework needs a different line ending (win32) we can convert it on that chrome side.

Reply:
This seems like a chrome issue rather than a LibWeb issue. AppKit and LibGUI already do break on `\r`.

AppKit:
<img width="182" alt="Screenshot 2024-03-08 at 7 20 37 AM" src="https://github.com/SerenityOS/serenity/assets/5600524/c92da665-49e8-42fe-9a49-74d4b431dca1">

LibGUI (though the height measurement is off):
![Screenshot_20240308_071813](https://github.com/SerenityOS/serenity/assets/5600524/a249d39b-9fde-4985-8fe6-b70ee4b434ce)

It seems only Qt does not break on CR, so if we need to break on CR, the fix here IMO should be in the Qt side.

(And side note, this patch makes it so we allocate the title string 3 times for the probably very rare chance that it contains a CR, which is a bit sad)

Reply:
You are right I moved the change to the Qt chrome, I saw that you also fixed CR support in LibGUI in the mean time nice!